File: exercises/binarySearch.py
from random import randrange
from math import ceil

data = []

for x in range(0, 10):
    randNum = randrange(0, 20)
    data.append(randNum)


def selectionSort(data):
    for x in range(0, len(data)):
        small = data[x]
        found = False
        for y in range(x, len(data)):
            if data[y] < small:
                small = data[y]
                found = True
                index = y
        if found:
            data[x], data[index] = small, data[x]


def binarySearch(data, num):
    low = 0
    high = len(data) - 1
    while high > low:
        mid = int(ceil((high + low) / 2))
        if data[low] == num or data[mid] == num or data[high] == num:
            return True
        else:
            if data[mid] > num:
                high = mid
                low += 1
            else:
                if data[mid] < num:
                    low = mid
                    high -= 1
                else:
                    return False


print(data)
selectionSort(data)
print(data)
while True:
    theInput = int(input(">> "))

    if binarySearch(data, theInput):
        print("Found")
    else:
        print("Not found")

File: exercises/binarySearch.js
const random = require("random");
const readline = require("readline");

let randNums = [];
for (let x = 0; x < 10; x++) {
    const randNum = random.int(0, 10);
    randNums.push(randNum);
}

let selectionSort = (randNums) => {
    for (let x = 0; x < randNums.length; x++) {
        let low = randNums[x];
        let shouldSwap = false;
        for (let y = x; y < randNums.length; y++) {
            let numAtY = randNums[y];
            if (low > numAtY) {
                low = numAtY;
                var index = y;
                shouldSwap = true;
            }
        }
        if (shouldSwap) {
            let temp = randNums[x];
            randNums[x] = low;
            randNums[index] = temp;
        }
    }
    return randNums;
};

let binarySearch = (sortedArray, num) => {
    console.log(sortedArray);
    let low = 0;
    let high = sortedArray.length - 1;
    while (low < high) {
        let mid = parseInt(Math.ceil((high + low) / 2));
        if (
            sortedArray[low] === num ||
            sortedArray[mid] === num ||
            sortedArray[high] === num
        ) {
            return true;
        } else {
            if (sortedArray[mid] > num) {
                high = mid;
                low++;
            } else if (sortedArray[mid] < num) {
                low = mid;
                high--;
            }
        }
    }
};

// Getting user input

const r1 = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let sortedArray = selectionSort(randNums);

r1.question(">> ", (num) => {
    if (binarySearch(sortedArray, num)) {
        console.log("Found");
        r1.close();
    } else {
        console.log("Not found");
        r1.close();
    }
});

r1.on("close", () => {
    process.exit(0);
});

// Making array of random numbers

// Sorting the numbers

// Searching the sorted array
